exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,563 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
+				Amount:    td.Quantity.Float64(),
+				Fee:       td.FeeAmount.Float64(),
+				Side:      td.Side,
+				Type:      oType,
+			}
+		}
+		resp, err := e.GetOrder(ctx, orderID, "")
+		if err != nil {
+			smartOrders, err := e.GetSmartOrderDetails(ctx, orderID, "")
+			switch {
+			case err != nil:
+				return nil, err
+			case len(smartOrders) == 0:
+				return nil, order.ErrOrderNotFound
+			case len(smartOrders) != 1:
+				return nil, fmt.Errorf("%w: too may smart orders returned", common.ErrInvalidResponse)
+			}
+			s := smartOrders[0]
+			oType, err := order.StringToOrderType(s.Type)
+			if err != nil {
+				return nil, err
+			}
+			return &order.Detail{
+				Side:          s.Side,
+				Pair:          s.Symbol,
+				Exchange:      e.Name,
+				Trades:        orderTrades,
+				OrderID:       s.ID,
+				TimeInForce:   s.TimeInForce,
+				ClientOrderID: s.ClientOrderID,
+				Price:         s.Price.Float64(),
+				QuoteAmount:   s.Amount.Float64(),
+				Date:          s.CreateTime.Time(),
+				LastUpdated:   s.UpdateTime.Time(),
+				Amount:        s.Quantity.Float64(),
+				Type:          oType,
+				Status:        orderStateFromString(s.State),
+				AssetType:     stringToAccountType(s.AccountType),
+			}, nil
+		}
+		oType, err := order.StringToOrderType(resp.Type)
 		if err != nil {
 			return nil, err
 		}
-		tradeHistory.Price = trades[i].Rate
-		tradeHistory.Amount = trades[i].Amount
-		tradeHistory.Total = trades[i].Total
-		tradeHistory.Fee = trades[i].Fee
-		orderInfo.Trades = append(orderInfo.Trades, tradeHistory)
-	}
-
-	resp, err := e.GetAuthenticatedOrderStatus(ctx, orderID)
-	if err != nil {
-		if len(orderInfo.Trades) > 0 { // on closed orders return trades only
-			if strings.Contains(err.Error(), "Order not found") {
-				orderInfo.Status = order.Closed
-			}
-			return &orderInfo, nil
+		return &order.Detail{
+			Price:                resp.Price.Float64(),
+			Amount:               resp.Quantity.Float64(),
+			AverageExecutedPrice: resp.AveragePrice.Float64(),
+			QuoteAmount:          resp.Amount.Float64(),
+			ExecutedAmount:       resp.FilledQuantity.Float64(),
+			RemainingAmount:      resp.Quantity.Float64() - resp.FilledAmount.Float64(),
+			Cost:                 resp.FilledQuantity.Float64() * resp.AveragePrice.Float64(),
+			Side:                 resp.Side,
+			Exchange:             e.Name,
+			OrderID:              resp.ID.String(),
+			ClientOrderID:        resp.ClientOrderID,
+			Type:                 oType,
+			Status:               orderStateFromString(resp.State),
+			AssetType:            stringToAccountType(resp.AccountType),
+			Date:                 resp.CreateTime.Time(),
+			LastUpdated:          resp.UpdateTime.Time(),
+			Pair:                 pair,
+			Trades:               orderTrades,
+			TimeInForce:          resp.TimeInForce,
+		}, nil
+	case asset.Futures:
+		fResults, err := e.GetFuturesOrderHistory(ctx, "", "", "", "", orderID, "", "", time.Time{}, time.Time{}, 0, 0)
```

Comment:
has no code coverage
If it does, please stop using `require.NoError` and use `assert.NoError` for multi-case tests